M2/cmake/build-libraries.cmake

```diff
@@ -180,7 +180,7 @@ ExternalProject_Add(build-bdwgc
   CMAKE_ARGS        -DCMAKE_INSTALL_PREFIX=${M2_HOST_PREFIX}
                     -DCMAKE_SYSTEM_PREFIX_PATH=${M2_HOST_PREFIX}
                     -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
-                    -DBUILD_SHARED_LIBS=${BUILD_SHARED_LIBS}
+                    -DBUILD_SHARED_LIBS=ON # for ForeignFunctions package
```

Comment:
Why is this forced on?

Reply:
`ForeignFunctions` has a `getMemory` function for allocating memory, which is very useful when your foreign function has a pointer as one of its arguments.  Under the hood, it calls either `GC_malloc` or `GC_malloc_atomic`.  Currently, this is implemented as a foreign function at top level, which means that we need libgc to be dynamically linked so we can use `dlsym`.

It could be implemented differently to allow for static linking by moving some of that code to ffi.d.  Would that be preferable?

Reply:
I think so? The ability to have a completely static build is always good (though there are currently a few obstacles using the CMake build that I don't understand).

Reply:
FWIW, I checked and I have the same problems statically linking bdwgc in the cmake build on the `development` branch, so not it's not an issue with the newer version of bdwgc or anything.